docs/Interoperator-authentication.md

```diff
@@ -0,0 +1,52 @@
+# Service Fabrik Inter-operator Authentication
+
+##  Abstract
+
+This document describes the basic and mTLS authentication supported by the Service Fabrik inter-operator broker.
+
+## Table of Content
+- [Service Fabrik Inter-operator Basic Architecture](#service-fabrik-inter-operator-basic-architecture)
+  - [Abstract](#abstract)
+  - [Table of Content](#table-of-content)
+  - [Context](#context)
+  - [BasicAuthentication](#basic-authentication)
+  - [mTLSAuthentication](#out-of-the-box-mtls-authentication)
+    - [CertificateIdentityVerification](#certificate-identity-verification)
+
+## Context
+The following two authentication mechanisms are supported for communicating with the brokers:
+  - Basic Authentication: validates the username and password.
+  - mTLS Authentication: validates the client certificate's subject pattern.
+
+## Basic Authentication
+To use Basic Authentication, the username and password should be provided during the registration of the broker.
+This is relevant for both subaccount-scoped registration done by smctl, and global registration done by the Cross Registration Service (xRS) API.
+
+The broker will use the credentials provided during registration to perform the authentication.
+
+In order to use basic auth, "secureIncomingConnections" flag must be set to false.
+
+## Out-Of-The-Box mTLS Authentication
+Interoperator broker currently supports the Out-Of-The-Box mTLS authentication.
+
+Nginx ingress has been configured to support the same, by adding the following auth-tls annotations:
+  - nginx.ingress.kubernetes.io/auth-tls-verify-client: "on"
+    Must be set to "on" to enable verification of client certificates. When this annotation is set to "on", it requests a client certificate that must be signed by a certificate that is included in the secret key ca.crt of the secret specified by "nginx.ingress.kubernetes.io/auth-tls-secret"
+  - nginx.ingress.kubernetes.io/auth-tls-secret: namespace/secretName
+    This secret must have a file named ca.crt containing the full Certificate Authority chain ca.crt that is enabled to authenticate against this Ingress.
+  - nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream: true
+    when set to true, passes the received certificates to the upstream server in the header ssl-client-cert. For mTLS auth, this annotation must be set to true.
```

Comment:
```suggestion
  - `nginx.ingress.kubernetes.io/auth-tls-verify-client: "on"` <br>
    Must be set to "on" to enable verification of client certificates. When this annotation is set to "on", it requests a client certificate that must be signed by a certificate that is included in the secret key ca.crt of the secret specified by "nginx.ingress.kubernetes.io/auth-tls-secret"
  - `nginx.ingress.kubernetes.io/auth-tls-secret: namespace/secretName` <br>
    This secret must have a file named ca.crt containing the full Certificate Authority chain ca.crt that is enabled to authenticate against this Ingress.
  - `nginx.ingress.kubernetes.io/auth-tls-pass-certificate-to-upstream: true` <br>
    when set to true, passes the received certificates to the upstream server in the header ssl-client-cert. For mTLS auth, this annotation must be set to true.
```
Minor formatting suggestion.
`<br>` helps to print its following sentence to a new line.